crates/ra_analysis/src/descriptors/mod.rs

```diff
@@ -31,17 +36,86 @@ impl FnDescriptor {
             node.syntax().text().to_string()
         };
 
+        if let Some((comment_range, docs)) = FnDescriptor::extract_doc_comments_for_fndef(node) {
+            let comment_range = comment_range.checked_sub(node.syntax().range().start()).unwrap();
+            let start = u32::from(comment_range.start()) as usize;
+            let end = u32::from(comment_range.end()) as usize;
```

Comment:
you'll be able to do `comment_range.start().to_usize`  after https://github.com/rust-analyzer/rust-analyzer/pull/169